apps/prairielearn/src/migrations/20250822211535_enrollments__joined_at__add.test.ts

```diff
@@ -0,0 +1,57 @@
+import { afterEach, assert, test } from 'vitest';
+import z from 'zod';
+
+import { loadSqlEquiv } from '@prairielearn/postgres';
+import * as sqldb from '@prairielearn/postgres';
+
+import * as helperCourse from '../tests/helperCourse.js';
+import { after, runAllMigrationsBefore, runAllMigrationsIncluding } from '../tests/helperDb.js';
+import { getOrCreateUser } from '../tests/utils/auth.js';
+
+const sql = loadSqlEquiv(import.meta.url);
+
+test('joined_at migration works correctly', async () => {
+  afterEach(async () => {
+    await after();
+  });
+  await runAllMigrationsBefore('20250822211535_enrollments__joined_at__add', {
+    drop: true,
+  });
+  await helperCourse.syncCourse();
```

Comment:
Nit, I love an abstraction and I can imagine a helper like this:

```ts
await testMigration({
  name: '20250822211535_enrollments__joined_at__add',
  beforeMigration: async () => {
    // ...
  },
  afterMigration: async () => {
    // ...
  }
});
```

It would do the following:

- `runAllMigrationsBefore`
- call `beforeMigration`
- `runAllMigrationsThrough`
- call `afterMigration`
- `runRemainingMigration`

Reply:
More devil's advocate:

- One could share state by returning it out of `beforeMigration` and passing it to `afterMigration`, TypeScript generics would make that type-safe.
- Dropping or not dropping could still be controlled by an option if needed.

My assertion is that the specifics here are hard enough to get right (they're already wrong in this PR) that we'd benefit from an abstraction and that the vast majority of migration tests would work with said abstraction (or at least, I can't think of one that wouldn't work with it).

Reply:
Ok, I'll try that out.

Reply:
I made this change in [0537de9](https://github.com/PrairieLearn/PrairieLearn/pull/12789/commits/0537de95dae7959e98f38f085a3944dcf442011b).